tests/test/scheduler/test_remote_mpi_worlds.cpp

```diff
@@ -83,45 +96,69 @@ TEST_CASE_METHOD(RemoteMpiTestFixture, "Test send across hosts", "[mpi]")
     }
 
     // Destroy worlds
+    senderThread.join();
     localWorld.destroy();
-    remoteWorld.destroy();
-
-    server.stop();
 }
 
 TEST_CASE_METHOD(RemoteMpiTestFixture,
-                 "Test collective messaging across hosts",
```

Comment:
These tests weren't using two different threads (not sure why they were passing). Given that I had to change them, I also separated them into smaller tests, easier to identify which fails.